src/databases/coursesDb.js

```diff
@@ -1,39 +1,75 @@
 const createError = require('http-errors');
 
-const { processDbResponse, snakelize } = require('../utils/dbUtils');
+const { processDbResponse } = require('../utils/dbUtils');
 const configs = require('../../configs');
 const knex = require('knex')(configs.db); // eslint-disable-line
 
+const COURSES_TABLE = 'courses';
+const COURSE_USERS_TABLE = 'course_users';
 /**
  * Get courses.
  *
  */
-const getCourses = async ({ page, limit }) => {
 
-  let pageSize = configs.coursesConfig.pageSize;
+const getCoursesByUser = async ({
+  userId,
+  page,
+  limit
+}) => knex(COURSE_USERS_TABLE)
+  .select()
+  .where({ user_id: userId })
+  .returning('*')
+  .offset(page)
+  .limit(limit)
+  .then(processDbResponse)
+  .then((response) => {
+    if (!response) {
+      throw new createError.NotFound('Courses not found');
+    }
+    if (!response.length) {
+      return [response];
+    }
+    return response;
+  });
 
-  return knex('courses')
+const getCourses = async ({
+  page,
+  limit
+}) => {
+  const { pageSize } = configs.coursesConfig.pageSize;
+  const offset = limit !== null && limit !== undefined ? limit : pageSize;
+  return knex(COURSES_TABLE)
     .select()
     .returning('*')
+    .offset(page)
+    .limit(offset)
```

Comment:
should not offset be setted in the upper line?

Reply:
I messed up that variable's name it should be limit not offset :P
I'll change it.